core/src/main/java/com/linecorp/armeria/server/http/auth/HttpBasicAuthService.java

```diff
@@ -0,0 +1,71 @@
+package com.linecorp.armeria.server.http.auth;
+
+import java.util.function.Function;
+import java.util.function.Predicate;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.base.Strings;
+
+import com.linecorp.armeria.common.http.HttpHeaderNames;
+import com.linecorp.armeria.common.http.HttpHeaders;
+import com.linecorp.armeria.common.http.HttpRequest;
+import com.linecorp.armeria.common.http.HttpResponse;
+import com.linecorp.armeria.server.Service;
+
+/**
+ * An {@link HttpAuthService} with <a href="https://en.wikipedia.org/wiki/Basic_access_authentication">HTTP basic authentication</a> support.
+ * */
+public class HttpBasicAuthService extends HttpAuthService<BasicToken> {
+
+    private static final Logger logger = LoggerFactory.getLogger(HttpBasicAuthService.class);
+
+    private static final Pattern AUTHORIZATION_HEADER_PATTERN = Pattern.compile(
+            "(?i)basic\\s+(?<decoded>\\S+)");
+
+    /**
+     * Creates a new HTTP basic authorization {@link Service} decorator using the specified {@code predicate}.
+     */
+    public static Function<Service<? super HttpRequest, ? extends HttpResponse>,
+            HttpBasicAuthService> newDecorator(Predicate<? super BasicToken> predicate) {
+        return service -> new HttpBasicAuthService(service, predicate);
+    }
+
+    HttpBasicAuthService(Service<? super HttpRequest, ? extends HttpResponse> delegate,
+                         Predicate<? super BasicToken> predicate) {
+        super(delegate, predicate);
+    }
+
+    @Override
+    public String toString() {
+        StringBuilder sb = new StringBuilder("HttpBasicAuthService(predicate:");
+        sb.append(predicate().hashCode());
+        sb.append(")");
+        return sb.toString();
+    }
+
+    /**
+     * Extracts {@link BasicToken} instance from given {@code req}. If {@code req} is invalid, returns null.
+     * */
+    @Override
+    protected BasicToken extractAuthToken(HttpHeaders headers) {
+        try {
+            String authorization = headers.get(HttpHeaderNames.AUTHORIZATION);
+            if (Strings.isNullOrEmpty(authorization)) {
+                throw new IllegalArgumentException("Authorization header does not exist");
```

Comment:
If we are not going to log the stack trace, we should not use an exception here because the instantiation of an exception is pretty expensive.